### PR TITLE
Add support for Server-Side Encryption to S3 Provider

### PR DESCRIPTION
**Pull request by davidski**
A small change to add support for encryption artifacts uploaded to AWS S3. Default to false to avoid 
breaking existing users. Docs are updated, but my rspec is very rusty. Would appreciate assistance/
mentoring to get a proper test created.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
Could you fix the commit message on https://github.com/travis-ci/dpl/pull/546/commits/9e2dd1c5552f5e669e3c2b0a2577fe243681fb22? The current one is not very descriptive. Thanks.

File: README.md [hunk 1]
```diff
@@ -319,6 +319,7 @@ For authentication you can also use Travis CI secure environment variable:
 * **region**: S3 Region. Defaults to us-east-1.
 * **upload-dir**: S3 directory to upload to. Defaults to root directory.
 * **storage-class**: S3 storage class to upload as. Defaults to "STANDARD". Other values are "STANDARD_IA" or "REDUCED_REDUNDANCY". Details can be found [here](https://aws.amazon.com/s3/storage-classes/).
+* **server-side-encryption**: When set to `true`, use S3 Server Side Encryption (SSE-AES256). Defaults to non-encrypted.
```
Comment (BanzaiMan) [inline README.md]: The default value is `false`.

Comment (davidski) [inline README.md]: Will do. Was a little premature on this PR. Sorry for the churn. :)

**Comment by davidski** [comment]
FYI - A passing deploy/build with this branch is at https://travis-ci.org/davidski/infrastructure-lambdas/builds/183591326

File: README.md (hunk 1 above)
Comment (davidski) [inline README.md]: Very sorry about the README. I did make the change, but it looks like I didn't push it to the branch. Doh! Thanks for taking care of that and merging, @BanzaiMan!

**Pull request merged**